File: with-lombok/src/test/java/TestCleanup.java
import java.io.IOException;
import java.io.StringWriter;

import lombok.Cleanup;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

class TestCleanup {

    @Test
    // throws IOException is mandatory …
    void shouldCleanResource() throws IOException {
        @Cleanup StringWriter out = new StringWriter();

        out.write("Hello Java");

        Assertions.assertThat(out.toString()).hasToString("Hello Java");
    }
}

File: with-kotlin/src/test/kotlin/TestCleanup.kt
import java.io.StringWriter
import kotlin.test.Test
import kotlin.test.assertEquals

class TestCleanup {

    @Test
    fun `should clean resource`() {
        val out = StringWriter()

        // Extension function 'use'
        out.use { it.write("Hello Kotlin") }

        assertEquals("Hello Kotlin", out.toString())
    }
}
